backend/canisters/identity/impl/src/updates/remove_identity_link.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{mutate_state, RuntimeState};
+use canister_tracing_macros::trace;
+use ic_cdk::update;
+use identity_canister::remove_identity_link::{Args, Response};
+
+#[update]
+#[trace]
+fn remove_identity_link(args: Args) -> Response {
+    mutate_state(|state| remove_identity_link_impl(args, state))
+}
+
+fn remove_identity_link_impl(args: Args, state: &mut RuntimeState) -> Response {
+    let caller = state.env.caller();
+    let auth_principal = state.data.user_principals.get_auth_principal(&caller);
```

Comment:
I think you can skip fetching this this because everything you need should be returned by `get_by_auth_principal`